core/codenvy-hosted-machine-authentication/src/test/java/com/codenvy/machine/authentication/server/MachineServiceAuthLinksInjectorTest.java

```diff
@@ -0,0 +1,128 @@
+import org.eclipse.che.dto.server.DtoFactory;
+import org.everrest.core.impl.uri.UriBuilderImpl;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.util.List;
+
+import static org.eclipse.che.api.machine.shared.Constants.TERMINAL_REFERENCE;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * @author Anton Korneta.
+ */
+@Listeners(MockitoTestNGListener.class)
+public class MachineServiceAuthLinksInjectorTest {
+    private static final String URI_BASE     = "http://localhost:8080";
+    private static final String API_ENDPOINT = URI_BASE + "/api";
+
+    @Mock
+    private ServiceContext         serviceContextMock;
+    @Mock
+    private MachineRuntimeInfoDto  machineRuntimeInfoDtoMock;
+    @Mock
+    private HttpJsonRequestFactory requestFactoryMock;
+    @Mock
+    private ServerDto              serverDtoMock;
+    private HttpJsonRequest        requestMock;
+
+    private MachineServiceAuthLinksInjector machineLinksInjector;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        final UriBuilder uriBuilder = new UriBuilderImpl();
+        uriBuilder.uri(URI_BASE);
+        requestMock  = mock(HttpJsonRequest.class, new SelfReturningAnswer());
+        when(requestFactoryMock.fromUrl(anyString())).thenReturn(requestMock);
+        machineLinksInjector = new MachineServiceAuthLinksInjector(API_ENDPOINT, requestFactoryMock);
+        when(serviceContextMock.getBaseUriBuilder()).thenReturn(uriBuilder);
+        when(serverDtoMock.getRef()).thenReturn(TERMINAL_REFERENCE);
+        when(serverDtoMock.getUrl()).thenReturn(URI_BASE);
+        when(machineRuntimeInfoDtoMock.getServers()).thenReturn(ImmutableMap.of("server", serverDtoMock));
+    }
+
+    @Test
+    public void shouldInjectTerminalLinkWithMachineToken() throws Exception {
+        final String machineToken = "machine12";
+        final MachineTokenDto tokenDto = DtoFactory.newDto(MachineTokenDto.class)
+                                                          .withMachineToken(machineToken);
+        final HttpJsonResponse responseMock = mock(HttpJsonResponse.class);
+        when(responseMock.asDto(MachineTokenDto.class)).thenReturn(tokenDto);
+        when(requestMock.request()).thenReturn(responseMock);
+        final MachineDto machineDto = DtoFactory.newDto(MachineDto.class)
+                                                .withId("id")
+                                                .withWorkspaceId("wsId")
+                                                .withRuntime(machineRuntimeInfoDtoMock);
+        final List<Link> links = Lists.newArrayList();
```

Comment:
for my culture, what is the need to delegate to guava ? it's to avoid <> ?
Lists.newArrayList();
vs
new ArrayList<>() ?